src/middlewares/validateDataWithZod.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Request, Response } from 'express';
+import type { LoginSchemaType } from 'src/lib/zodSchema/loginSchema';
+import type { SignUpSchemaType } from 'src/lib/zodSchema/signUpSchema';
+
+const validateDataWithZod =
+  (Schema: LoginSchemaType | SignUpSchemaType) =>
```

Comment:
Can you make the Schema type generic by using the ZodSchema type to accept all zod schema types.

Reply:
It's not impossible